### ISSUE TITLE
Suggest Hoffman approximations for Fresnel factor of metal under coat/film

### ISSUE THREAD

**Issue by portsmouth** [AcademySoftwareFoundation/OpenPBR#225]
When metal is coated by a dielectric (or a thin film), its Fresnel factor will be modified due to the adjacent dielectric IOR differing from air.

Modeling this in OpenPBR is currently not clearly stated since our metal lobe uses the "F82-tint" model, which is based on the Schlick approximation, where don't know the underlying IOR and absorption of the metal (needed to compute the Fresnel factor correctly).

So we currently just suggest to use the Gulbrandsen approach to calculate this (we mention this only in the thin-film section, but I'd also suggest we move this to the Metal section, as it applies in the case of a coated metal also):

>In the case of the metallic base the physics is somewhat ambiguous since, as described in the [Metal](https://academysoftwarefoundation.github.io/OpenPBR/#metal) section, the Fresnel factor for metal is defined according to the Schlick-based “F82-tint” parametrization which does not specify the underlying physical complex IOR. We suggest here that some reasonable approximation is employed to map the Fresnel factor to the best matching effective complex IOR, for example that described by [[Gulbrandsen2014](https://academysoftwarefoundation.github.io/OpenPBR/#citation-gulbrandsen2014)].

However, Naty Hoffman (@natyh) (who developed the F82-tint model) has suggested some approximations to compute the modified Fresnel of metal under dielectric, within the F82-tint model. As he noted, this could be just as (or more) accurate than working with the true Fresnel equations, in the context of an RGB renderer.

His formulas are given in his talk at the SIGGRAPH 2020 [course ](https://blog.selfshadow.com/publications/s2020-shading-course/)"Physically Based Shading in Theory and Practice":

https://blog.selfshadow.com/publications/s2020-shading-course/hoffman/s2020_pbs_hoffman_slides.pdf

His first suggestion is to just use the IORs alone (inferred from the F0) to do the calculation, which he shows works reasonably well:

![image](https://github.com/user-attachments/assets/e0ac7cbf-e76b-4af5-830f-56767823c87a)

His more accurate suggestion is the following formula (which isn't derived, so we should ask where this comes from):

![image](https://github.com/user-attachments/assets/c75bbf38-2842-4364-8dad-53bbe45bb917)

**Comment by portsmouth** [comment]
Related to this,  I was thinking that a more principled way to describe the metal would be to specify the required (observed) F0 and F82 reflectances, but leave the specific underlying metal model that achieves this implementation dependent and ideally physically accurate. 

So that would amount to:
   - specify the F0 color via `base_color`
   - specify the F82 color,  which we could still do using the tint parametrization, i.e. F82 is given by `specular_color` times the Schlick curve for the specified F0.

That then defines the physical RGB F0 and F82. Then we would say that there is assumed to be some underlying physical metal that generates those reflectances (in the given color space), i.e. some spectral IOR $n$ and absorption $k$. That would then completely define how the interaction with the adjacent media should happen. 

There would in general be many possible $n(\lambda)$ and $k(\lambda)$ that achieve this, and any one will do. That leaves it open for future improvements to provide a good approximation (in some sense) for the underlying spectral metal that produces the desired reflection colors. 

We can then just suggest the Lazanyi-Schlick model as a convenient fit that hits the required colors. (And the Hoffman formulas quotes here for external media handling).

**Comment by noj86** [comment]
There's been many conversation on this topic for previous public projects.
([https://github.com/Autodesk/standard-surface/issues/12](https://github.com/Autodesk/standard-surface/issues/12))

To this day and from my perspective this publication

[https://blog.selfshadow.com/publications/s2020-shading-course/belcour/slides/index.html#/
](https://blog.selfshadow.com/publications/s2020-shading-course/belcour/slides/index.html#/)
[https://belcour.github.io/blog/research/publication/2020/08/26/brdf-fresnel-decompo.html](https://belcour.github.io/blog/research/publication/2020/08/26/brdf-fresnel-decompo.html)

is a stronger solution to the metallic fresnel than the current F82 parameter, particularly in a context where openpbr offers a metalness control which would allow a transition from a full dielectric fresnel equation to a full conductor fresnel. More than the benefit on the metal, I see a huge value in representing all non-metallic surface with a full dielectric equation over a schlick term.

**Comment by natyh** [comment]
I disagree on metallic Fresnel, but the point on the advantages of dielectric Fresnel is a good one that has been in the back of my mind for some years now. Re-thinking this a bit, maybe the best approach is to use both - full dielectric Fresnel for metalness 0, the Schlick-Lazanyi-F82-tint model for metalness 1, and a simple mix / blend between the two for intermediate metalness values? The approach (which I have been advocating until now) of using a single Schlick-Lazanyi-F82-tint lobe to cover all metalness values is slightly cheaper to compute (which was an especially strong consideration in my mind given my real-time rendering background), but upon further consideration that may not be as big an issue as I was thinking. For many real-time applications the computation cost difference is insignificant, and (given the rather subtle nature of edge dip effects) the more performance-critical ones (mobile, VR, etc.) will probably just use regular Schlick regardless of what OpenPBR ends up using.
